yellow-paper/docs/public-vm/nested-calls.mdx

```diff
@@ -0,0 +1,110 @@
+# Nested Contract Calls
+
+A **nested contract call** occurs _during_ AVM execution and is triggered by a **contract call instruction** ([`CALL`](./instruction-set#isa-section-call), [`STATICCALL`](./instruction-set#isa-section-staticcall), or [`DELEGATECALL`](./instruction-set#isa-section-delegatecall)). It initializes a new execution context (**nested context**) from the current one (**calling context**) and triggers execution within it. When nested call's execution completes, execution proceeds in the calling context.
+
+## Nested Call Instructions
+
+The AVM supports three types of contract calls: [`CALL`](./instruction-set#isa-section-call), [`STATICCALL`](./instruction-set#isa-section-staticcall), and [`DELEGATECALL`](./instruction-set#isa-section-delegatecall).
+
+These call instructions share the same argument definitions: `gasOffset`, `addrOffset`, `argsOffset`, `argsSize`, `retOffset`, `retSize`, and `successOffset`. These arguments will be referred to via those keywords below, and will often be used in conjunction with the `M[offset]` syntax which is shorthand for `context.machineState.memory[offset]`.
+
+The terms `isStaticCall` and `isDelegateCall` used below simply refer to the call instruction's type (`instr.opcode == STATICCALL`, `instr.opcode == DELEGATECALL`).
+
+## Tracing nested contract calls
+
+Before nested execution begins, the contract call is traced.
+```jsx
+context.worldStateAccessTrace.contractCalls.append(
+    TracedContractCall {
+        callPointer: context.worldStateAccessTrace.contractCalls.length + 1,
+        address: M[addrOffset],
+        storageAddress: M[addrOffset],
+        counter: context.worldStateAccessTrace.accessCounter++,
```

Comment:
`context.worldStateAccessTrace.accessCounter++` could be slightly ambiguous. In C++ this means "first return accessCounter, then increment"* (same in TS/JS), so the counter value in the struct would be the one without the increment. 

*that's the diff between ++i and i++ and why it's recommended to use ++i when it doesn't matter. The compiler doesn't need to save the value of i in a separate register/mem.